File: ClientApp/src/app/model/message-requests/MessageRequest.ts
export class MessageRequest {
    constructor(public roomId: string, public message: string, public asAnonymous: boolean, public iv: string, public type: string, public messageId?: string) { }

    static createWithId(roomId: string, message: string, asAnonymous: boolean, messageId: string, iv: string, type: string): MessageRequest {
        return new MessageRequest(roomId, message, asAnonymous, messageId, iv, type);
    }

    static createWithoutId(roomId: string, message: string, asAnonymous: boolean, iv: string, type: string): MessageRequest {
        return new MessageRequest(roomId, message, asAnonymous, iv, type);
    }
}

File: Server/MessagesServer/Model/Requests/Message/MessageRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MessagesServer.Model.Requests.Message;

public record MessageRequest(
    [Required(ErrorMessage = "Room id cannot be null.")]string RoomId,
    [Required(ErrorMessage = "Message cannot be null.")]string Message,
    [Required(ErrorMessage = "'AsAnonym' bool cannot be null.")]bool AsAnonymous,
    [Required(ErrorMessage = "Iv cannot be null.")]string Iv,
    string? MessageId);